src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -562,19 +567,82 @@ internal void Detach(bool removeAsync, bool removePrimary)
 
             var filteredContextData = new Dictionary<string, object>();
 
-            // first, handle the "includes"
-            if (_configurationService.Configuration.ContextDataInclude.Count() == 0) // empty "include" list, so include everything
+            // Pseudocode
```

Comment:
There's enough specialized code here that you might want to split it out into a new class/file.

Reply:
Done.